Mathlib/LinearAlgebra/DirectSum/Finsupp.lean

```diff
@@ -24,36 +58,134 @@ open DirectSum
 
 open Set LinearMap Submodule
 
-variable {R : Type u} {M : Type v} {N : Type w} [Ring R] [AddCommGroup M] [Module R M]
-  [AddCommGroup N] [Module R N]
+variable {R : Type u} {M : Type v} {N : Type w}
+  [CommSemiring R] [AddCommMonoid M] [Module R M] [AddCommMonoid N] [Module R N]
 
 section TensorProduct
 
 open TensorProduct
 
-open TensorProduct
+variable {ι : Type*} [DecidableEq ι]
+
+/-- The tensor product of `i →₀ M` and `N` is linearly equivalent to `i →₀ M ⊗[R] N` -/
+noncomputable def Finsupp.rTensor :
```

Comment:
Arguably this should be called `TensorProduct.finsuppLeft` for consistency with `TensorProduct.directSumLeft` and `TensorProduct.prodLeft`, and similarly for `Right`

Reply:
(I had to check for the actual meaning of “Arguably” ! — it doesn't mean one can argue, and indeed, i won't…)